### DISCUSSION TITLE
Syncing pre-existing files

### DISCUSSION THREAD

**Discussion by Tippon** [abraunegg/onedrive#2186] [Q&A] [answered]
First of all, thanks for creating onedrive. It's going to make a big difference in helping me switch to Linux full time :)

If I copy my OneDrive files from my PC to my laptop, do I need to do anything different when I run onedrive on the laptop, or do I just run it with the --synchronize flag?

I've got over 900GB of files to sync, so I'm going to put the laptop drive into the PC to transfer the files first, or it will take several days to download them.

Thanks :)

**Comment by abraunegg** [comment, marked as answer]
@Tippon 
> If I copy my OneDrive files from my PC to my laptop, do I need to do anything different when I run onedrive on the laptop, or do I just run it with the --synchronize flag?

You should not need to do anything different - the client should detect that the files etc are the same, and just update the local cache database to record that everything is in sync.

Before you run the application however - please make sure that:
* Your Windows | Linux system are using the same Timezone
* Your Windows | Linux systems are using NTP and successfully synced
* You sync from Windows -> OneDrive online to ensure OneDrive online has the correct sync state of your files.

This will ensure that when the Linux client checks the files, they are of the same size, same timestamps, thus will not attempt to re-upload or change anything online.

Use `--dry-run` to validate and test as well.

**Reply by Tippon** [reply]
That's great, thank you :)

The times and time zones are correct, and Windows should be fully synced, so I'll double check that and start the copy.
